### PR TITLE
Refactor xhr/resources/inspect-headers.py to avoid six

### PR DESCRIPTION
**Pull request by foolip**
This is used by tests in xhr/ and one in html/:
html/webappapis/system-state-and-capabilities/the-navigator-object/navigator.any.js

See also https://github.com/web-platform-tests/wpt/issues/28756.

### CODE REVIEW

**Review by annevk — COMMENTED** [review]
So it seems that we basically regress with Python 3 in what we can test. I think I noted this when we did the Python 3 migration as well, but I somehow thought it was addressed.

Browsers actually had differences in how they serialized an HTTP/1 request and that's why it carefully checks for say, a space following `:`.

I guess there's no alternative and we effectively already lost this coverage, but let's at least put a comment here about what we'd ideally want to do and maybe also link an issue.

**Comment by foolip** [comment]
I see, so this wasn't merely about working around str/bytes changes, but the original code gave us the headers block as a single buffer that we could look at, is that right?

**Comment by annevk** [comment]
Yeah, indeed.

**Comment by annevk** [comment]
I filed https://github.com/web-platform-tests/wpt/issues/28756. If you update the comment to point there I'm happy with landing this.

File: xhr/resources/inspect-headers.py [hunk 1]
```diff
@@ -1,32 +1,15 @@
...
-            return b"Syntax error: missing CRLF: " + line
-        line = line[:-2]
-
-        if b': ' not in line:
-            return b"Syntax error: no colon and space found: " + line
-        name, value = line.split(b': ', 1)
-
+    # raw_headers.raw_items() returns the (name, value) header pairs as
+    # tuples of strings. Convert them to bytes before comparing.
+    # TODO: Get access to the raw headers, so that whitespace between
```
Comment (foolip) [inline xhr/resources/inspect-headers.py:7]: @annevk does the comment capture it?

**Review by annevk — APPROVED** [review]
Yeah this seems fine. I suspect we want to check CRLF as well, but I don't think that ever caught an actual issue.

**Pull request merged**